core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -726,17 +725,7 @@ default Optional<Type> getSupportedType(ConnectorSession session, Map<String, Ob
      */
     default Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
-        ConnectorTableProperties properties = getTableProperties(session, tableHandle);
```

Comment:
will this break insert scaling for existing Trino connectors?

Reply:
This method is implemented for Hive, Iceberg and Delta.
This is not a change for these connectors.

getTableProperties() is additionally implemented for Phoenix, Kudu, Mongo, Elasticsearch, OpenSearch, Sheets
but without returning partitioning properties, so this is not a change for these connectors.

getTableProperties() is implemented and can return partitioning properties for TPCH
but there is no insert support, so this is not a change for this connector.

getTableProperties() is implemented and returns partitioning for Raptor, so this is a change for this connector. I will update Raptor.

Last but not least, this is a SPI breaking change, so it may break connectors that are not shipped with Trino.
I don't think there is a way to fix the ConnectorMetadata API without such breakage, but i can make the breakage more obvious -- throw when the code is invoked and *would do* something.

Reply:
the fact that ConnectorMetadata provides default implementation for insertLayout which is not necessarily good for every connector (just a heuristic) and is not consistent with default implementation for other `get*Layout` methods